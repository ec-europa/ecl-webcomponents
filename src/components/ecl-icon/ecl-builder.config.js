const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-ec-icons/dist/sprites/icons.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-eu-icons/dist/sprites/icon.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
