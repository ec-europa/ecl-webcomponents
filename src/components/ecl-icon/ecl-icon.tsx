import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}