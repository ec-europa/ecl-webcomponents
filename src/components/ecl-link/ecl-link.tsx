import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-link',
  styleUrls: {
    ec: './build/styles/ecl-link-ec.css',
    eu: './build/styles/ecl-link-eu.css',
  },
  shadow: false,
  scoped: false,
  assetsDirs: ['build'],
})

export class EclLink {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() titleAttr: string;
  @Prop() external: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-link`,
      `ecl-link--${this.variant}`,
      this.styleClass
    ];

    if (this.el.querySelector('ecl-icon')) {
      styleClasses.push('ecl-link--icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        styleClasses.push('ecl-link--icon-after');
      } else {
        styleClasses.push('ecl-link--icon-before');
      }
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('ecl-icon')) {
      this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--after');
      } else {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--before');
      }
    }
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class={`ecl-link__icon ecl-link__icon--after sc-ecl-link-${this.theme}`}
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  render() {
    return (
      <a
        class={this.getClass()}
        href={this.path}
        title={this.titleAttr}
      >
        <slot name="icon-before"></slot>
        <slot></slot>
        <slot name="icon-after">{ this.external ? this.getExternal() : '' }</slot>
      </a>
    );
  }
}
