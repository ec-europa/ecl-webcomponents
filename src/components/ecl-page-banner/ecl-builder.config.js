const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-page-banner-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-page-banner-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-page-banner-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-page-banner-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
