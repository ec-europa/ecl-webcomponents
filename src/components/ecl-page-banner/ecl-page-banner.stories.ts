const getArgs = () => {
  return {
    variant: 'primary',
    bannerTitle: 'EU Budget for the future',
    description: 'Innovation, economy, environment and geopolitics',
    ctaLabel: 'Subscribe',
    ctaLink: '/example.html',
    image: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg',
    fullWidth: false,
    centered: true,
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: [
        'primary',
        'secondary',
        'white',
        'image',
        'image-gradient',
        'image-shade'
      ],
      description: "Page banner variants"
    },
    bannerTitle: {
      name: 'title',
      type: { name: 'string' },
      description: "Title of the banner"
    },
    description: {
      type: { name: 'string' },
      description: 'Banner description',
    },
    image: {
      type: { name: 'string' },
      description: 'Image path or url',  
    },
    ctaLabel: {
      name: 'label of the CTA button',
      type: { name: 'string' },
      description: 'CTA Link label',  
    },
    ctaLink: {
      name: 'link of the CTA link',
      type: { name: 'string' },
      description: 'Link path or url',  
    },
    fullWidth: {
      name: 'full width banner',
      type: { name: 'boolean' },
      description: 'Extends the banner to the viewport size',
    },
    centered: {
      type: { name: 'boolean' },
      description: 'Centered banner content',
    },
  };
};

export default {
  title: 'Components/page banner',
};

const Template = (args) => 
`<ecl-page-banner
  variant="${args.variant}"
  theme="${args.theme}"
  cta-link="${args.ctaLink}"
  cta-label="${args.ctaLabel}"
  image="${args.image}"
  banner-title="${args.bannerTitle}"
  credit="Â© Copyright or credit"
>
  ${args.description}
</ecl-page-banner>`;

export const PageBanner = Template.bind({});
PageBanner.storyName = 'default';
PageBanner.args = getArgs();
PageBanner.argTypes = getArgTypes();
