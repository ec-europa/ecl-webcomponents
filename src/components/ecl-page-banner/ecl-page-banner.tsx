import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-page-banner',
  styleUrls: {
    ec: './build/styles/ecl-page-banner-ec.css',
    eu: './build/styles/ecl-page-banner-eu.css',
  },
  shadow: false,
  scoped: true,
})
export class EclPageBanner {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() external: boolean = false;
  @Prop() variant: string = 'display';
  @Prop() fullWidth: boolean = false;
  @Prop() bannerTitle: string;
  @Prop() image: string;
  @Prop() ctaLabel: string;
  @Prop() ctaLink: string;
  @Prop() centered: boolean = true;
  @Prop() credit: string;

  getClass(): string {
    const styleClasses = [
      `ecl-page-banner`,
      `ecl-page-banner--${this.variant}`,
      this.styleClass
    ];

    if (this.centered) {
      styleClasses.push('ecl-page-banner--centered');
    }
    if (this.fullWidth) {
      styleClasses.push('ecl-page-banner--full-width');
    }

    return styleClasses.join(' ');
  }

  render() {
    const imageBanners = [
      'image',
      'image-gradient',
      'image-shade',
    ];
    return (
      <section class={this.getClass()}>
      { imageBanners.includes(this.variant) && this.image ?
        <div
          class="ecl-page-banner__image"
          style={{ backgroundImage: "url(" + this.image + ")"}}
        >
        </div>
        : '' }
        { this.credit && imageBanners.includes(this.variant) ?
        <div class="ecl-page-banner__credit">{this.credit}</div> : '' }
        <div class="ecl-container">
          <div class="ecl-page-banner__container">
            <div class="ecl-page-banner__content">
              <div class="ecl-page-banner__title">{this.bannerTitle}</div>
              <p class="ecl-page-banner__description">
                <span class="ecl-page-banner__description-text">
                  <slot></slot>
                </span>
              </p>
            { this.ctaLabel && this.ctaLink ?
              <div class="ecl-page-banner__cta">
                <ecl-link
                  variant="cta"
                  path={this.ctaLink}
                  styleClass="ecl-page-banner__link-cta sc-ecl-page-banner-ec"
                >
                  {this.ctaLabel}
                  <ecl-icon
                    icon="corner-arrow"
                    slot="icon-after"
                    transform="rotate-90"
                  ></ecl-icon>
                </ecl-link>
              </div>
            : '' }
            </div>
          </div>
        </div>
      </section>
    );
  }
}
