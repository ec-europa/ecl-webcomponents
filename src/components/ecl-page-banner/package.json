{
  "name": "@ecl/web-components-page-banner",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.9.0",
  "description": "ECL Page Banner",
  "main": "ecl-page-banner.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.9.1",
    "@ecl/vanilla-component-page-banner": "3.9.1",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
