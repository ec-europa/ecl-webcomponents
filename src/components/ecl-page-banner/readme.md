# ecl-page-banner



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description | Type      | Default     |
| ------------- | -------------- | ----------- | --------- | ----------- |
| `bannerTitle` | `banner-title` |             | `string`  | `undefined` |
| `centered`    | `centered`     |             | `boolean` | `true`      |
| `credit`      | `credit`       |             | `string`  | `undefined` |
| `ctaLabel`    | `cta-label`    |             | `string`  | `undefined` |
| `ctaLink`     | `cta-link`     |             | `string`  | `undefined` |
| `external`    | `external`     |             | `boolean` | `false`     |
| `fullWidth`   | `full-width`   |             | `boolean` | `false`     |
| `image`       | `image`        |             | `string`  | `undefined` |
| `styleClass`  | `style-class`  |             | `string`  | `undefined` |
| `theme`       | `theme`        |             | `string`  | `'ec'`      |
| `variant`     | `variant`      |             | `string`  | `'display'` |


## Dependencies

### Used by

 - [ecl-carousel-item](../ecl-carousel)

### Depends on

- [ecl-link](../ecl-link)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-page-banner --> ecl-link
  ecl-page-banner --> ecl-icon
  ecl-link --> ecl-icon
  ecl-carousel-item --> ecl-page-banner
  style ecl-page-banner fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
