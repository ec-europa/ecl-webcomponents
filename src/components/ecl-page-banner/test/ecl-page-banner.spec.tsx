import { newSpecPage } from '@stencil/core/testing';
import { EclPageBanner } from '../ecl-page-banner';
import { EclLink } from '../../ecl-link/ecl-link';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-page-banner', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclPageBanner, EclLink, EclIcon],
      html: `<ecl-page-banner
              variant="primary"
              theme="ec"
              cta-link="/example.html"
              cta-label="Subscribe"
              image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
              banner-title="EU Budget for the future"
            >
              Innovation, economy, environment and geopolitics
            </ecl-page-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
