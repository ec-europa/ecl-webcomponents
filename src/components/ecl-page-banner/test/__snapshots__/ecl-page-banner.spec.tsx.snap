// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-page-banner renders 1`] = `
<ecl-page-banner banner-title="EU Budget for the future" credit="Â© Copyright or credit" cta-label="Subscribe" cta-link="/example.html" image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" theme="ec" variant="primary">
  <!---->
  <section class="ecl-page-banner ecl-page-banner--centered ecl-page-banner--primary">
    <div class="ecl-container">
      <div class="ecl-page-banner__container">
        <div class="ecl-page-banner__content">
          <div class="ecl-page-banner__title">
            EU Budget for the future
          </div>
          <p class="ecl-page-banner__description">
            <span class="ecl-page-banner__description-text">
              Innovation, economy, environment and geopolitics
            </span>
          </p>
          <div class="ecl-page-banner__cta">
            <ecl-link style-class="ecl-page-banner__link-cta sc-ecl-page-banner-ec">
              <!---->
              <a class="ecl-link ecl-link--cta ecl-link--icon ecl-link--icon-after ecl-page-banner__link-cta sc-ecl-page-banner-ec" href="/example.html">
                Subscribe
                <slot-fb hidden="" name="icon-after"></slot-fb>
                <ecl-icon slot="icon-after">
                  <svg class="ecl-icon ecl-icon--rotate-90 ecl-icon--xs ecl-link__icon ecl-link__icon--after">
                    <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
                  </svg>
                </ecl-icon>
              </a>
            </ecl-link>
          </div>
        </div>
      </div>
    </div>
  </section>
</ecl-page-banner>
`;
